### PR TITLE
[freerdp] install dependencies from nightly control

### PR DESCRIPTION
**Pull request by ligurio**
* install apt dependencies from nightly package control file
* get CMake configuration from toolchain file

### CODE REVIEW

**Comment by DonggeLiu** [comment]
Thanks @ligurio!
Let me convert this into a draft for now, but please feel free to convert it back when you have a chance to address the CI errors : )

File: projects/freerdp/Dockerfile [hunk 1]
```diff
@@ -17,19 +17,23 @@
...
 
 RUN git clone --depth 1 https://github.com/FreeRDP/FreeRDP.git
 
 WORKDIR FreeRDP
+
+# Install all dependencies required by the nightly package.
+RUN sed -i 's/# deb-src/deb-src/' /etc/apt/sources.list
+RUN apt update
+RUN apt-get install -y apt-cudf aspcud
+RUN apt-cache showsrc FreeRDP/packaging/deb/freerdp-nightly/control
```
Comment (akallabeth) [inline projects/freerdp/Dockerfile:32]: you should already be in the source directory here, right?

Comment (ligurio) [reply projects/freerdp/Dockerfile:32]: Why full path to `control` is not enough?

Comment (akallabeth) [reply projects/freerdp/Dockerfile:32]: you´re not using full path, but a wrong one.
you are already in `FreeRDP` so full path should be `packaging/deb/freerdp-nightly/control`

Comment (akallabeth) [reply projects/freerdp/Dockerfile:32]: see `WORKDIR FreeRDP` above

**Comment by akallabeth** [comment]
@ligurio ok, looks like it does not like the `ICU` library, we can deactivate that in the ci preload file on our end.

File: projects/freerdp/build.sh [hunk 2]
```diff
@@ -77,7 +60,7 @@ cmake_args=(
 
 # Build the project and fuzzers.
 rm -rf build
-cmake "${cmake_args[@]}" -S . -B build -G Ninja
+cmake "${cmake_args[@]}" -S . -B build -G Ninja -C ci/cmake-preloads/config-oss-fuzz.cmake
```
Comment (ligurio) [inline projects/freerdp/build.sh:63]:
```
reference to `__gxx_personality_v0'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libicuuc.a(umutex.ao): in function `umtx_init':
(.text+0x3b): undefined reference to `std::condition_variable::condition_variable()'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libicuuc.a(umutex.ao): in function `umtx_cleanup':
(.text+0xc0): undefined reference to `std::condition_variable::~condition_variable()'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libicuuc.a(umutex.ao): in function `icu_66::umtx_initImplPostInit(icu_66::UInitOnce&)':
(.text+0x178): undefined reference to `std::__throw_system_error(int)'
/usr/bin/ld: (.text+0x17f): undefined reference to `std::__throw_system_error(int)'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libicuuc.a(umutex.ao): in function `icu_66::umtx_initImplPreInit(icu_66::UInitOnce&)':
(.text+0x25b): undefined reference to `std::condition_variable::wait(std::unique_lock<std::mutex>&)'
/usr/bin/ld: (.text+0x2be): undefined reference to `std::__throw_system_error(int)'
/usr/bin/ld: (.text+0x2ca): undefined reference to `std::__throw_system_error(int)'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libicuuc.a(umutex.ao): in function `icu_66::UMutex::getMutex()':
(.text+0x398): undefined reference to `std::__throw_system_error(int)'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libicuuc.a(umutex.ao): in function `umtx_lock_66':
(.text+0x3ea): undefined reference to `std::__throw_system_error(int)'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libicuuc.a(umutex.ao): in function `icu_66::umtx_initImplPostInit(icu_66::UInitOnce&)':
(.text+0x16e): undefined reference to `std::condition_variable::notify_all()'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libicuuc.a(umutex.ao): in function `std::call_once<void (&)()>(std::once_flag&, void (&)())::{lambda()#2}::_FUN()':
(.text._ZZSt9call_onceIRFvvEJEEvRSt9once_flagOT_DpOT0_ENUlvE0_4_FUNEv[_ZZSt9call_onceIRFvvEJEEvRSt9once_flagOT_DpOT0_ENUlvE0_4_FUNEv]+0x7): undefined reference to `std::__once_callable'
```

@akallabeth any ideas?

Comment (akallabeth) [reply projects/freerdp/build.sh:63]: @ligurio see above. you can do a test by adding `-DWITH_UNICODE_BUILTIN=ON`. if that works, we´ll add it to the ci script on our end.

File: projects/freerdp/build.sh [hunk 3]
```diff
@@ -73,11 +56,13 @@ cmake_args=(
...
     -DCMAKE_MODULE_LINKER_FLAGS="${LDFLAGS}"
     -DCMAKE_SHARED_LINKER_FLAGS="${LDFLAGS}"
+
+    -DWITH_UNICODE_BUILTIN=ON
 )
 
 # Build the project and fuzzers.
 rm -rf build
-cmake "${cmake_args[@]}" -S . -B build -G Ninja
+cmake "${cmake_args[@]}" -S . -B build -G Ninja -C ci/cmake-preloads/config-oss-fuzz.cmake
```
Comment (ligurio) [inline projects/freerdp/build.sh:65]:
```
/tmp/not-out/tmp6xge54sh/TestFuzzCommonAssistanceParseFileBuffer: error while loading shared libraries: libopus.so.0: cannot open shared object file: No such file or directory
```
@akallabeth

Comment (ligurio) [reply projects/freerdp/build.sh:65]: `-DWITH_OPUS=OFF`?

File: projects/freerdp/build.sh (hunk 2 above)
Comment (ligurio) [reply projects/freerdp/build.sh:66]: Fixed.

File: projects/freerdp/build.sh (hunk 3 above)
Comment (ligurio) [reply projects/freerdp/build.sh:66]: Fixed.

File: projects/freerdp/Dockerfile (hunk 1 above)
Comment (ligurio) [reply projects/freerdp/Dockerfile]: Fixed.

**Comment by ligurio** [comment]
Waiting for https://github.com/FreeRDP/FreeRDP/pull/10093

**Review by akallabeth — APPROVED** [review]
LGTM, except for the `UNICODE_BUILTIN` commit which should be removed before merging.

**Review by akallabeth — APPROVED** [review]
LGTM

**Comment by akallabeth** [comment]
@ligurio thx for the help!

**Review by DonggeLiu — APPROVED** [review]

**Pull request merged**